File: Sort-Evaluation/Source/Sort.cpp
#include <bits/stdc++.h>
#include <chrono>
#include <fstream>
#include <random>
#define fastio ios_base::sync_with_stdio(0), cin.tie(0);

using namespace std;
using namespace chrono;

vector <int> readFile(const string& filename){
    ifstream file(filename);
    vector <int> data;
    int val;

    if(file.is_open()){
        while(file >> val){
            data.push_back(val);
        }
        file.close();
    }
    else {
        cerr << "Unable to open file " << filename << '\n';
    }
    return data;
}


int main()
{
    vector<string> filenames;

    for(int i = 1; i <= 10; i++){
        filenames.push_back("../Make_Test/test" + to_string(i) + ".inp");
    }

    freopen("Sort.txt", "w", stdout);
    for(const string& filename : filenames){
        vector<int> data = readFile(filename);
        auto start = high_resolution_clock::now();
        sort(data.begin(), data.end());
        auto stop = high_resolution_clock::now();
        cout << duration_cast<milliseconds>(stop - start).count() << '\n';
    }
    return 0;
}

File: Sort-Evaluation/Source/Sort.py
import numpy as np
import time
import os

def read_input_file(file_path):
    with open(file_path, "r") as file:
        data = file.read().strip().split()
    return [int(float(x)) for x in data]


input_files = [ f"../Make_Test/test{i}.inp" for i in range (1, 11)]

for file_path in input_files:
    arr = read_input_file(file_path)
    arr = np.array(arr)

    start = time.time()
    np.sort(arr)
    stop = time.time()

    delta_time = (stop - start) * 1000

    print(f"{delta_time:.2f}")


